test/integration/tests/metagov.ts

```diff
@@ -0,0 +1,457 @@
+        await contracts.core
+          .connect(daoSigner)
+          .revokeRole(ethers.utils.id('METAGOVERNANCE_TOKEN_STAKING'), deployAddress);
+      });
+
+      it('initial lock()', async function () {
+        expect(await contracts.angle.balanceOf(locker.address)).to.be.equal(e18(10_000_000));
+        expect(await contracts.veAngle.balanceOf(locker.address)).to.be.equal('0');
+        await locker.lock();
+        expect(await contracts.angle.balanceOf(locker.address)).to.be.equal('0');
+        expect(await contracts.veAngle.balanceOf(locker.address)).to.be.at.least(e18(9_900_000));
+      });
+      it('2 years later, increase lock time', async function () {
+        await time.increase(2 * 365 * 86400); // 2 years
+        expect(await contracts.angle.balanceOf(locker.address)).to.be.equal('0');
+        expect(await contracts.veAngle.balanceOf(locker.address)).to.be.at.least(e18(4_900_000));
+        expect(await contracts.veAngle.balanceOf(locker.address)).to.be.at.most(e18(5_100_000));
+        await locker.lock();
+        expect(await contracts.angle.balanceOf(locker.address)).to.be.equal('0');
+        expect(await contracts.veAngle.balanceOf(locker.address)).to.be.at.least(e18(9_900_000));
+      });
+      it('lock more tokens', async function () {
+        const holder = '0x4f91F01cE8ec07c9B1f6a82c18811848254917Ab';
+        await forceEth(holder);
+        const signer = await getImpersonatedSigner(holder);
+        await contracts.angle.connect(signer).transfer(locker.address, e18(1_000_000));
+        await locker.lock();
+        expect(await contracts.angle.balanceOf(locker.address)).to.be.equal('0');
+        expect(await contracts.veAngle.balanceOf(locker.address)).to.be.at.least(e18(10_900_000));
+      });
+      it('fast-forward 4 years', async function () {
+        // fast-forwarding 4 years directly causes a revert without reason,
+        // so we fast-forward 1 year at a time, and perform a checkpoint()
+        // regularly.
+        await time.increase(1 * 365 * 86400); // 1 years
+        await contracts.veAngle.checkpoint();
+        await time.increase(1 * 365 * 86400); // 2 years
+        await contracts.veAngle.checkpoint();
+        await time.increase(1 * 365 * 86400); // 3 years
+        await contracts.veAngle.checkpoint();
+        await time.increase(1 * 365 * 86400); // 4 years
+        await contracts.veAngle.checkpoint();
+      });
+      it('exitLock() at the end of lockup period', async function () {
+        expect(await contracts.angle.balanceOf(locker.address)).to.be.equal('0');
+        expect(await contracts.veAngle.balanceOf(locker.address)).to.be.equal('0');
+        await locker.exitLock();
+        expect(await contracts.angle.balanceOf(locker.address)).to.be.at.least(e18(10_900_000));
+        expect(await contracts.veAngle.balanceOf(locker.address)).to.be.equal('0');
+      });
+    });
+  });
+
+  describe('utils/OZGovernorVoter.sol', function () {
```

Comment:
I don't like nesting multiple different test blocks into the same file. Why not create a parent `metagov` directory and then put each of these `describe` blocks into their own file? It would split things up more + reduce the amount of code to review in each file.

If some `before` block code is needed across each `describe` then we can break it out into a `fixture` and import

Reply:
sounds good, I'll split the files